src/coreclr/vm/ceeload.h

```diff
@@ -1706,7 +1706,7 @@ class Module
 
         if (pLoadLevel && !th.IsNull())
         {
-            if (!IsCompilationProcess() && (flags & ZAPPED_TYPE_NEEDS_NO_RESTORE))
+            if (flags & ZAPPED_TYPE_NEEDS_NO_RESTORE)
```

Comment:
`ZAPPED_TYPE` would indicate NGen in some manner. Are we going to leave this for a future PR or can we verify it is no longer used and remove here?

Reply:
I was going to do a separate pass to deal with `MethodTable::IsPreRestored` and these flags.